querybook/webapp/components/DataDocDAGExporter/DataDocDAGExporter.tsx

```diff
@@ -21,47 +25,85 @@ export const DataDocDAGExporter: React.FunctionComponent<IProps> = ({
+    const [isExporting, setIsExporting] = React.useState(false);
+    const [exportData, setExportData] = React.useState<string>();
+    const isInteractive = !(readonly || isExporting);
+
+    const { onSave, savedNodes, savedEdges, savedMeta } = useSavedDAG(docId);
+    const queryCells = useQueryCells(docId);
+    const [nodes, edges, setNodes, setEdges, dropRef] = useExporterDAG(
+        queryCells,
+        savedNodes,
+        savedEdges,
+        !isInteractive
+    );
+    const unusedQueryCells = useUnusedQueryCells(queryCells, nodes);
 
-    const {
-        deleteGraphQueryCell,
-        unusedQueryCells,
-        graphQueryCells,
-        dropRef,
-    } = useGraphQueryCells(docId, savedNodes, readonly);
+    const handleExport = React.useCallback(
+        async (exporterName: string, exporterSettings: Record<string, any>) => {
+            const meta = { ...savedMeta, [exporterName]: exporterSettings };
+            await onSave(nodes, edges, meta);
+
+            const { data: exportData } = await DataDocResource.exportDAG(
+                docId,
+                exporterName,
+                nodes,
+                edges,
+                exporterSettings
+            );
+            setExportData(exportData);
+            return exportData;
+        },
+        [docId, nodes, edges, onSave, savedMeta]
+    );
 
     return (
         <div className="DataDocDAGExporter">
-            <DataDocDagExporterList queryCells={unusedQueryCells} />
-            <div className="DataDocDAGExporter-main">
-                <div className="DataDocDAGExporter-graph-wrapper" ref={dropRef}>
-                    <DataDocDAGExporterGraph
-                        queryCells={graphQueryCells}
-                        savedNodes={savedNodes}
-                        savedEdges={savedEdges}
-                        onDeleteCell={deleteGraphQueryCell}
-                        renderSaveComponent={
-                            readonly
-                                ? null
-                                : (nodes, edges) => (
-                                      <DataDocDAGExporterSave
-                                          onSave={() => onSave(nodes, edges)}
-                                      />
-                                  )
-                        }
-                        readonly={readonly}
-                    />
-                </div>
-            </div>
+            {isExporting ? (
+                <DataDocDAGExporterForm
+                    handleExport={handleExport}
+                    savedMeta={savedMeta}
+                    dropRef={dropRef}
+                    nodes={nodes}
+                    edges={edges}
+                    setNodes={setNodes}
+                    setEdges={setEdges}
+                    onSave={onSave}
+                    onReturn={() => setIsExporting(false)}
+                    clearExportData={() => setExportData(undefined)}
+                    exportData={exportData}
```

Comment:
nit: you can let DAGExporter handle rendering of exportData
```
{
    exportData && (
        <Modal onHide={() => setExportData(undefined)>{exportData}</Modal>
     )
}
```
also to handle exportType, you can add a param for handleExport to get exportType, and let exportData store that as well. Alternatively, you can make the API return the type with the data